src/core/hle/D3D8/Direct3D9/PixelShader.cpp

```diff
@@ -219,12 +219,20 @@ void FinalCombinerStageHlsl(std::stringstream& hlsl, RPSFinalCombiner& fc)
 	hlsl << "\n	" << opcode_comment[5][0] << "(" << arguments.str() << "); // " << opcode_comment[5][1];
 }
 
-void OutputDefine(std::stringstream& hlsl, std::string define_str, bool enabled)
+void OutputDefineFlag(std::stringstream& hlsl, bool enabled, std::string define_enabled, std::string define_disabled = "")
```

Comment:
```suggestion
void OutputDefineFlag(std::stringstream& hlsl, bool enabled, std::string_view define_enabled, std::string_view define_disabled = "")
```

If stringstream supports string views, of course.

Reply:
interesting... I guess string almost never needs to be passed then?

Reply:
Yes! It only makes sense when you want to "own" it, e.g. in constructors. For all other cases, `string_view` is superior because it is a near no-op conversion from both `std::string` and char literals.

The only "price" you pay is lack of a null terminator, but if you don't need one then you can use `string_view`.